src/hooks/useFeedbackDisclosure.ts

```diff
@@ -0,0 +1,66 @@
+import { useDisclosure } from "@chakra-ui/react"
+import { useLocation } from "react-router-dom"
+
+import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
+
+import { useLoginContext } from "contexts/LoginContext"
+
+import { isEditPageUrl, isSpecialPagesUrl } from "utils/pages"
+
+import { useLocalStorage } from "./useLocalStorage"
+
+type LastSeenFeedbackTime = number
+type UserId = string
+
+type FeedbackStorageMappings = Record<UserId, LastSeenFeedbackTime>
+
+const NPS_SURVEY_INTERVAL = 1814400000 as const // 3 weeks - 3 * 7 * 24 * 60 * 60 * 1000
+
+const { REACT_APP_SHOW_NPS_FORM } = process.env
+
+// NOTE: Wrapper to handle get/set of individual keys
+const useFeedbackStorage = () => {
+  const { userId } = useLoginContext()
+  const [
+    userMappings,
+    setUserMappings,
+  ] = useLocalStorage<FeedbackStorageMappings>(LOCAL_STORAGE_KEYS.Feedback, {})
+
+  const lastSeen = userMappings[userId]
+  const setLastSeen = () => {
+    setUserMappings({ ...userMappings, [userId]: Date.now() })
+  }
+  // NOTE: `const` cast to infer types properly
+  // so that it knows that the return value is a tuple
+  return [lastSeen, setLastSeen] as const
+}
+
+export const useFeedbackDisclosure = () => {
+  const location = useLocation<{ from?: string }>()
+  // NOTE: Despite the typing from the library,
+  // state is NOT guaranteed to exist.
+  const from = location.state?.from ?? ""
+  // NOTE: We show the feedback modal to the users iff
```

Comment:
super nit: typo, change `iff` to `if`

Reply:
iff is intentional, it means if and only if